lib/extensions/ar_region.rb

```diff
@@ -117,6 +117,18 @@ def partition_objs_by_remote_region(objs)
     def partition_ids_by_remote_region(ids)
       ids.partition { |id| self.id_in_current_region?(id) }
     end
+
+    private
+
+    def discover_my_region_number
+      id_to_region(connection.select_value("select last_value from miq_databases_id_seq"))
```

Comment:
I'm not familiar with how this query works.

How would this work with a replication scenario where you have rows:
1
1 trillion 1
10 trillion 1

With local REGION saying 10?
With local REGION saying 1?
With local REGION saying 0?
No REGION file?